pkg/logs/tail.go

```diff
@@ -0,0 +1,603 @@
+		// customer wants to consume.
+		// Undefined == both stderr and stdout.
+		stream string
+	}
+
+	// Log defines the message envelope that compute@edge (C@E) wraps the
+	// user messages in.
+	Log struct {
+		// SequenceNum is the message sequence number used to reorder
+		// messages.
+		SequenceNum int `json:"sequence_number"`
+		// RequestTime is the time in microseconds when the request
+		// was received.
+		RequestStart int64 `json:"request_start_us"`
+		// Stream is the C@E stream, either stdout or stderr.
+		Stream string `json:"stream"`
+		// RequestID is a UUID representing individual requests to the
+		// particular WASM service.
+		RequestID string `json:"id"`
+		// Message is the actual message body the user wants printed.
+		Message string `json:"message"`
+	}
+
+	// Batch encompasses a batch ID and the logs for this batch.
+	Batch struct {
+		ID   string `json:"batch_id"`
+		Logs []Log  `json:"logs"`
+	}
+)
+
+// NewTailCommand returns a usable command registered under the parent.
+func NewTailCommand(parent common.Registerer, globals *config.Data) *TailCommand {
+	var c TailCommand
+	c.Globals = globals
+	c.CmdClause = parent.Command("tail", "Tail Compute@Edge logs")
+	c.CmdClause.Flag("service-id", "Service ID").Short('s').StringVar(&c.manifest.Flag.ServiceID)
+	c.CmdClause.Flag("from", "From time, in unix seconds").Int64Var(&c.cfg.from)
+	c.CmdClause.Flag("to", "To time, in unix seconds").Int64Var(&c.cfg.to)
+	c.CmdClause.Flag("sort-buffer",
+		"Sort buffer is how long to buffer logs, attempting to sort them before printing, defaults to 1s (second)").Default("1s").DurationVar(&c.cfg.sortBuffer)
+	c.CmdClause.Flag("search-padding",
+		"Search padding is how much of a window on either side of From and To to use for searching, defaults to 2s (seconds)").Default("2s").DurationVar(&c.cfg.searchPadding)
+	c.CmdClause.Flag("stream", "Stream specifies which of 'stdout' or 'stderr' to output, defaults to undefined (all streams)").StringVar(&c.cfg.stream)
+
+	return &c
+}
+
+// Exec invokes the application logic for the command.
+func (c *TailCommand) Exec(in io.Reader, out io.Writer) error {
+	serviceID, source := c.manifest.ServiceID()
+	if source == manifest.SourceUndefined {
+		return errors.ErrNoServiceID
+	}
+	c.Input.ServiceID = serviceID
+
+	c.Input.Kind = fastly.ManagedLoggingInstanceOutput
+	c.cfg.path = fmt.Sprintf("%s/service/%s/log_stream/managed/instance_output", config.DefaultEndpoint, c.Input.ServiceID)
+
+	c.dieCh = make(chan struct{})
+	c.batchCh = make(chan Batch)
+	c.doneCh = make(chan struct{})
+
+	c.hClient = http.DefaultClient
+	c.token, _ = c.Globals.Token()
+
+	// Adjust the from/to times if they are
+	// defined. We adjust the times based on searchPadding.
+	c.adjustTimes()
+
+	// Enable managed logging if not already enabled.
+	if err := c.enableManagedLogging(out); err != nil {
+		return err
+	}
+
+	sigs := make(chan os.Signal, 2)
+	signal.Notify(sigs, os.Interrupt, syscall.SIGTERM, syscall.SIGHUP)
```

Comment:
OK so this: https://go-review.googlesource.com/c/go/+/187739 << suggests that SIGTERM and SIGINT are both already ok to use then when compiled for Windows. It looks like they get internally re-mapped to the posix equivalent in the signal handler. So we should be OK to swap to syscall for a bit more consistency:
```suggestion
	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM, syscall.SIGHUP)
```

But I'm also happy for this to be left as is if you prefer not to discover some strange edge case.

Reply:
Fixed